yargs.js

```diff
@@ -523,6 +523,7 @@ function Yargs (processArgs, cwd, parentRequire) {
 
       var demand = opt.demand || opt.required || opt.require
 
+      // deprecated, use 'demandOption' instead
       if (demand) {
         self.demand(key, demand)
```

Comment:
@JaKXz I've been bit a couple times where I have a dependency of a dependency that starts printing out ugly warning messages, that are a pain in the butt to get rid of in the output.

I was thinking, with yargs, it might be worth deprecating in documentation without actually spitting out warning messages.